pkg/api/validation/validation.go

```diff
@@ -351,6 +352,15 @@ func ValidateImmutableField(newVal, oldVal interface{}, fldPath *field.Path) fie
 	return allErrs
 }
 
+func ValidateImmutableAnnotation(newVal, oldVal interface{}, annotation string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if !api.Semantic.DeepEqual(oldVal, newVal) {
```

Comment:
annotation values are strings, no need for generic `interface{}` and `DeepEqual` here. Use `string` and simple `==`.